katalog/etcd-backup/etcd-backup-s3/cronjob.yaml

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2017-present SIGHUP s.r.l All rights reserved.
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: etcd-backup-s3
+spec:
+  schedule: "* * 2 * *"
+  successfulJobsHistoryLimit: 3
+  failedJobsHistoryLimit: 5
+  concurrencyPolicy: Allow
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          volumes:
+            - name: etcd-certs
+              hostPath:
+                path: /etc/etcd/pki/etcd
+            - name: persistence
+              emptyDir: {}
+            - name: host-backup
+              hostPath:
+                path: /tmp/etcd-backup
+            - name: etcd-backup-s3-rclone-conf
+              secret:
+                secretName: etcd-backup-s3-rclone-conf
+          initContainers:
+            - name: backupper
+              image: etcd
+              envFrom:
+                - configMapRef:
+                    name: etcd-backup-certificates-location
+              securityContext:
+                runAsUser: 0
+                runAsGroup: 0
```

Comment:
Is there a reason why we are running this as root? We only need to run etcdctl cli

Reply:
Yes, there is a reason. We are `hostPath` mounting the `healthcheck` certificates on the controlplanes in order to talk with the etcd server. Those are usually owned by the root user on the host filesystem, and thus it is necessary to be root inside the container to read them.

Reply:
Test `fsGroup` here

Reply:
Just tested and `fsGroup` does not help here. Moreover, setting `fsGroup` would change permission (not in our case, since this is implemented in a per-plugin basis) to the volume to match the user set. Again, this is not our case, since for the time being the hostPath plugin does not support `chown`ing the volume to the `fsGroup` set in the manifest, but it wouldn't be a great idea to change permissions to the certificates.